libs/utils/include/utils/memalign.h

```diff
@@ -99,6 +99,7 @@ class STLAlignedAllocator {
         aligned_free(p);
     }
 
+	#ifndef _MSC_VER
```

Comment:
need a comment that explains why we remove this for msvc.

Reply:
Because when these are defined I get the following error:

```
1>JobSystem.cpp
1>C:\greg\github\filament\libs\utils\include\utils/memalign.h(106,5): error C2995:  'bool utils::operator ==(const utils::STLAlignedAllocator<TYPE> &,const utils::STLAlignedAllocator<U> &)': function template has already been defined
1>C:\greg\github\filament\libs\utils\include\utils/memalign.h(106): message :  see declaration of 'utils::operator =='
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(628): message :  see reference to class template instantiation 'utils::STLAlignedAllocator<T>' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(628): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(628): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(628): message :             T=std::_Container_proxy
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(628): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\vector(625): message :  while compiling class template member function 'std::vector<utils::JobSystem::ThreadState,utils::STLAlignedAllocator<utils::JobSystem::ThreadState>>::~vector(void) noexcept'
1>C:\greg\github\filament\libs\utils\src\JobSystem.cpp(111): message :  see reference to function template instantiation 'std::vector<utils::JobSystem::ThreadState,utils::STLAlignedAllocator<utils::JobSystem::ThreadState>>::~vector(void) noexcept' being compiled
1>C:\greg\github\filament\libs\utils\include\utils/JobSystem.h(400): message :  see reference to class template instantiation 'std::vector<utils::JobSystem::ThreadState,utils::STLAlignedAllocator<utils::JobSystem::ThreadState>>' being compiled
1>C:\greg\github\filament\libs\utils\include\utils/memalign.h(112,5): error C2995:  'bool utils::operator !=(const utils::STLAlignedAllocator<TYPE> &,const utils::STLAlignedAllocator<U> &)': function template has already been defined
1>C:\greg\github\filament\libs\utils\include\utils/memalign.h(112): message :  see declaration of 'utils::operator !='
```

Reply:
It is not clear to me where these operators==() and !=() are already defined though??

Reply:
It seems related to this code in the std::vector implementation:

![image](https://user-images.githubusercontent.com/1496907/65546412-772fa900-dee5-11e9-8f59-7b75e41ad1b0.png)

Reply:
Error occurs only when _ITERATOR_DEBUG_LEVEL != 0.

The following code builds file:

```
#if !defined(_MSC_VER) || _ITERATOR_DEBUG_LEVEL == 0 // msvc std::vector debug iterator code seems to somehow redefine these operators
    // stateless allocators are always equal
    template<typename T, typename U>
    friend bool
    operator==(const STLAlignedAllocator<T>& rhs, const STLAlignedAllocator<U>& lhs) {
        return true;
    }

    template<typename T, typename U>
    friend bool
    operator!=(const STLAlignedAllocator<T>& rhs, const STLAlignedAllocator<U>& lhs) {
        return false;
    }
#endif
```

Reply:
I checked in that latest version